### ISSUE TITLE
Title-case for uppercase strings (textcase_CapitalsUntouched)

### ISSUE THREAD

**Issue by larsgw** [citation-style-language/test-suite#9]
The specs say, about [title case conversion](http://docs.citationstyles.org/en/stable/specification.html#title-case-conversion):

> 1. For uppercase strings, the first character of each word remains capitalized. All other letters are lowercased.

Yet in the following test case, two uppercase strings ("UK" and "OC 1") are capitalized according to rule 2 (where non-lowercase words stay the same):

https://github.com/citation-style-language/test-suite/blob/d72a1ebb959dc67e8b1c6dbfc2e6a8be934ecb68/processor-tests/humans/textcase_CapitalsUntouched.txt#L7-L13

**Comment by fbennett** [comment]
Yes, the behavior doesn't fit the stated rules here, does it. it's reflecting behaviour of `citeproc-js`, which I think is coded to avoid touching things that look like acronyms (OECD, WTO, WHO, FAA...), which might well turn up as the sole "word" in a title.

**Comment by larsgw** [comment]
What things look like acronyms? How would one distinguish between a book titled Dr Who but written in uppercase and a website about Disaster Relief by the World Health Organization? I guess this is an edge case, and I don't have an incentive for objecting to the fixture, I was just wondering.

**Comment by fbennett** [comment]
It take that back. It's not guessing at acronym-like things, it's doing what the name of the test suggests: capital letters are not lowercased. That makes the behaviour even further removed from the spec language, but also simpler to explain. It also fits the advice commonly given to users by the CSL team: to set titles in sentence case, capitalizing only things (like acronyms and proper nouns) that should always be in capitals. It's not one for my desk, but this would be a case for amendment of the spec language, I think.

**Comment by bdarcus** [comment]
Sounds like this is another issue that would benefit from a PR against the spec?

**Comment by larsgw** [comment]
I think the amendment would be removing point 1 from the spec entirely, I don't know if that's wanted. But I can make a PR later.

**Comment by bdarcus** [comment]
To be clear, I haven't looked at the specifics closely. I'm just trying to try, across a number of repos, and gathered that from what I could tell of where the conversation ended up.

Thoughts @fbennett?

**Comment by adam3smith** [comment]
Removing point 1 (and adjusting the remaining phrasing) seems right to me.
This removes the ability for title-case to fix poor data entry, but it avoids the risk of unintended lowercasing of an all acronym (or otherwise intentionally all caps) title. This seems like the right approach to me -- I'm not aware of any complaints about the current citeproc-js behavior.

**Comment by rmzelle** [comment]
> Removing point 1 (and adjusting the remaining phrasing) seems right to me.
> This removes the ability for title-case to fix poor data entry, but it avoids the risk of unintended lowercasing of an all acronym (or otherwise intentionally all caps) title. This seems like the right approach to me -- I'm not aware of any complaints about the current citeproc-js behavior.

I have a bunch of old items in my Zotero library with all-caps title (mostly patents like https://patentscope.wipo.int/search/en/detail.jsf?docId=WO2011142027), but it looks like Zotero nowadays sentence cases them when saving. (plus Zotero has the option to re-case titles post-save)

I'd be fine with the tenet that CSL processors shouldn't try to clean up messy metadata, which is the main purpose of the described behavior for all-uppercase strings. (although we might want to mention that somewhere)
